azurerm/internal/services/eventhub/eventhub_namespace_dedicated_resource.go

```diff
@@ -0,0 +1,416 @@
+											string(eventhub.NetworkRuleIPActionAllow),
+										}, false),
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+
+			"default_primary_connection_string_alias": {
+				Type:      schema.TypeString,
+				Computed:  true,
+				Sensitive: true,
+			},
+
+			"default_secondary_connection_string_alias": {
+				Type:      schema.TypeString,
+				Computed:  true,
+				Sensitive: true,
+			},
+
+			"default_primary_connection_string": {
+				Type:      schema.TypeString,
+				Computed:  true,
+				Sensitive: true,
+			},
+
+			"default_primary_key": {
+				Type:      schema.TypeString,
+				Computed:  true,
+				Sensitive: true,
+			},
+
+			"default_secondary_connection_string": {
+				Type:      schema.TypeString,
+				Computed:  true,
+				Sensitive: true,
+			},
+
+			"default_secondary_key": {
+				Type:      schema.TypeString,
+				Computed:  true,
+				Sensitive: true,
+			},
+
+			"tags": tags.Schema(),
+		},
+	}
+}
+
+func resourceArmeventHubNamespaceDedicatedCreateUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Eventhub.NamespacesClient
+	ctx, cancel := timeouts.ForCreateUpdate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+	log.Printf("[INFO] preparing arguments for AzureRM EventHub Namespace creation.")
+
+	name := d.Get("name").(string)
+	clusterID := d.Get("cluster_id").(string)
+	resGroup := d.Get("resource_group_name").(string)
+
+	if features.ShouldResourcesBeImported() && d.IsNewResource() {
+		existing, err := client.Get(ctx, resGroup, name)
+		if err != nil {
+			if !utils.ResponseWasNotFound(existing.Response) {
+				return fmt.Errorf("Error checking for presence of existing EventHub Namespace %q (Resource Group %q): %s", name, resGroup, err)
+			}
+		}
+
+		if existing.ID != nil && *existing.ID != "" {
+			return tf.ImportAsExistsError("azurerm_eventhub_namespace", *existing.ID)
+		}
+	}
+
+	location := azure.NormalizeLocation(d.Get("location").(string))
+	sku := d.Get("sku").(string)
+	capacity := int32(d.Get("capacity").(int))
+	t := d.Get("tags").(map[string]interface{})
+	autoInflateEnabled := d.Get("auto_inflate_enabled").(bool)
+
+	parameters := eventhub.EHNamespace{
+		Location: &location,
+		Sku: &eventhub.Sku{
+			Name:     eventhub.SkuName(sku),
+			Tier:     eventhub.SkuTier(sku),
+			Capacity: &capacity,
+		},
+		EHNamespaceProperties: &eventhub.EHNamespaceProperties{
+			IsAutoInflateEnabled: utils.Bool(autoInflateEnabled),
+			ClusterArmID:         &clusterID,
+		},
+		Tags: tags.Expand(t),
+	}
+
+	if v, ok := d.GetOk("maximum_throughput_units"); ok {
+		parameters.EHNamespaceProperties.MaximumThroughputUnits = utils.Int32(int32(v.(int)))
+	}
+
+	future, err := client.CreateOrUpdate(ctx, resGroup, name, parameters)
+	if err != nil {
+		return err
+	}
+
+	if err = future.WaitForCompletionRef(ctx, client.Client); err != nil {
+		return fmt.Errorf("Error creating eventhub namespace: %+v", err)
```

Comment:
again for clarity on resource type, we should specify the dedicated nature in this message:
```suggestion
		return fmt.Errorf("Error creating dedicated eventhub namespace: %+v", err)
```